### DISCUSSION TITLE
Support for Jakarta annotations?

### DISCUSSION THREAD

**Discussion by craig-day** [square/retrofit#3757] [General]
Would it make sense to migrate to, or just add support for, the [Jakarta EE annotations for creating service interfaces](https://jakarta.ee/specifications/restful-ws/3.0/apidocs/jakarta/ws/rs/package-summary.html)? 

These examples should be equivalent.

Current: 
```java
interface MyService {

  @GET("path/to/object/{id}")
  Object object(@Path("id") String id);
}
```
Jakarta: 
```java
interface MyService {

  @GET
  @Path("/path/to/object/{id}")
  Object object(@PathParam("id") String id);
}

**Comment by JakeWharton** [comment]
I looked at JAX-RS 1 in around 2012 and decided not to support it, and then I looked at JAX-RS 2 back in 2015 and decided not to support it (https://github.com/square/retrofit/issues/671#issuecomment-79939297). I didn't realize there was a version 3 and have not looked to see what it offers compared to the other versions.

Generally, we don't feel like we want to adhere to a spec. I believe Retrofit is/was successful because we got to build the best client-side version of what we wanted by only serving our needs and the needs of our users directly.

The example you show is very minimal and the scope and complexity of our annotations and the RS ones both go up but not necessarily in the same direction. Retrofit doesn't actually support your example as written as you would have to return a `Call<Object>`. In order to return a naked `Object` you would need to write a custom `CallAdapter.Factory` (which is possible).

**Comment by craig-day** [comment]
Makes sense. Sorry I didn't see that issue and duplicated it here. Regarding `Object`, that was just a placeholder in the method signature.

**Reply by JakeWharton** [reply]
It's a very old issue so no worries. I wish I had written more about the investigation then but often these things are done quickly and we don't write up as much as we should.
